src/main/java/org/terasology/tutorialWorldGeneration/HouseRasterizer.java

```diff
@@ -47,14 +47,15 @@ public void generateChunk(CoreChunk chunk, Region chunkRegion) {
             Vector3i centerHousePosition = new Vector3i(entry.getKey());
             int extent = entry.getValue().getExtent();
             centerHousePosition.add(0, extent, 0);
-            BlockRegion walls = BlockRegion.createFromCenterExtents(centerHousePosition, extent);
-            BlockRegion inside = BlockRegion.createFromCenterExtents(centerHousePosition, extent - 1);
+            BlockRegion walls = new BlockRegion(centerHousePosition);
+            BlockRegion inside = new BlockRegion(walls).expand(extent - 1, extent - 1, extent - 1);
+            walls.expand(extent, extent, extent);
 
             // loop through each of the positions in the cube, ignoring the is
             for (Vector3ic newBlockPosition : walls) {
                 if (chunkRegion.getRegion().contains(newBlockPosition)
                         && !inside.contains(newBlockPosition)) {
-                    chunk.setBlock(Chunks.toRelative(newBlockPosition), stone);
+                    chunk.setBlock(Chunks.toRelative(new Vector3i(newBlockPosition)), stone);
```

Comment:
As these tutorial modules should not only explain the concept, but also display good/best practice use. Thefore, let's talk math 👨‍🏫 

`chunk.setBlock(...)` does take a position for read-only access, i.e., it does not store the instance you pass in. We can make use of this property to only having a single instance of `Vector3i` that we modify in-place in the loop. As `Chunks.toRelative` (with single argument) also modifies the vector in-place, we cannot pass in the `newBlockPosition` loop variable. 

A typical pattern that we applied in many places already is to use a single temporary vector instance to be reused in the loop:

```java
Vector3i tmp = new Vector3i();
for (Vector3ic newBlockPosition : walls) {
	if (chunkRegion.getRegion().contains(newBlockPosition)
    		&& !inside.contains(newBlockPosition)) {
    	chunk.setBlock(Chunks.toRelative(newBlockPosition, tmp), stone);
}
```

Here, we also make use of the two-argument variant of `Chunks.toRelative` which will store the result in the second argument `dest` - in this case our `tmp` variable.